src/main/java/uk/ac/ebi/embl/api/validation/check/feature/CollectionDateQualifierCheck.java

```diff
@@ -85,57 +85,67 @@ public ValidationResult check(Feature feature)
 		for (Qualifier collectionQualifier : collectionDateQualifiers) 
 		{
 			String value = collectionQualifier.getValue();
-			if (value == null || value.isEmpty())
-			{
+			if (isValueEmpty(value)) {
 				reportError(collectionQualifier.getOrigin(),
-					DATE_FORMAT_ERROR, collectionQualifier.getName(), "");
+						DATE_FORMAT_ERROR, collectionQualifier.getName(), "");
 			}
-						
-			String valueNoWhitespace = StringUtils.deleteWhitespace(value);
 
-			if (valueNoWhitespace == null || valueNoWhitespace.isEmpty())
-			{
-				reportError(collectionQualifier.getOrigin(),
-					DATE_FORMAT_ERROR, collectionQualifier.getName(), "");
-			}	
-		
-			if (!valueNoWhitespace.equals(value))
-			{
-				 collectionQualifier.setValue(valueNoWhitespace);
-				 value = valueNoWhitespace;
+			value = StringUtils.deleteWhitespace(value);
+			collectionQualifier.setValue(value);
+			try {
+				if (!isValidDate(value)) {
+					reportError(collectionQualifier.getOrigin(), DATE_FORMAT_ERROR, collectionQualifier.getName(), value);
+				}
+			} catch (IllegalArgumentException e) {
+				reportError(collectionQualifier.getOrigin(), FUTURE_DATE_ERROR, collectionQualifier.getName(), collectionQualifier.getValue());
 			}
+		}
+		return result;
+	}
 
-			Matcher m = DATE_RANGE_PATTERN.matcher(value);
-			
-			if (m.matches())
-			{
-				CheckDateResult fromResult = checkDate(collectionQualifier, m.group(1));
-				CheckDateResult toResult = checkDate(collectionQualifier, m.group(2));				
-				
-				if (fromResult.date != null && toResult.date != null)
-				{
-					if (fromResult.pattern != toResult.pattern)
-					{
-						// Different range date formats.
-						reportError(collectionQualifier.getOrigin(),
-							DATE_FORMAT_ERROR, collectionQualifier.getName(), value);
-					}
-					if (fromResult.date.compareTo(toResult.date) > 0)
-					{
-						// From date in a range is larger than to date.						
-						reportError(collectionQualifier.getOrigin(),
-								DATE_FORMAT_ERROR, collectionQualifier.getName(), value);						
-					}
+	public boolean isValid(final String value) {
```

Comment:
- Note to myself: ```isValid(final String value)``` has been added to validate value extracted from Sample XML so that we don't inject invalid values.
- Suggestion: add interface:
```
/** Some source feature qualifiers originating from samples are validated before they are added to the source feature so that only valid qualifier values are added.
*/
public interface isValidSourceFeatureQualifier() {
    boolean isValidSourceFeatureQualifier(final String value);
}
```